pytreex/core/node.py

```diff
@@ -321,13 +327,19 @@ def create_child(self, id=None, data=None):
         return getattr(sys.modules[__name__],
                        self.__class__.__name__)(data=data, parent=self)
 
-    def remove(self):
+    def remove(self, fix_order=True):
         "Remove the node from the tree."
+        root = self.root # backup, self.root will not be reliable (why?)
         for child in self.get_children():
-            child.remove()
+            child.remove(fix_order=False)
         self.parent = None
         self.document.remove_node(self.id)
 
+        # We need to normalize ordering, so there are no gaps
```

Comment:
[Yes](https://github.com/ufal/treex/blob/master/lib/Treex/Core/Node.pm#L303-L305).
It could be done more efficiently (update just the following ords).
Also if you delete several nodes at once, you can use `fix_order` only with the last call for speedup (this is not possible in Perl Treex yet).